### PR TITLE
docker and docker-compose: Updates to file paths to clean up dirs (Fix #266)

### LINKED ISSUE #266: Unable to run web image
See below:
```
$ kubectl logs -n reach web-6ff6f6d7fd-gwzft
[2019-11-05 14:07:33 +0000] [1] [INFO] Starting gunicorn 19.9.0
[2019-11-05 14:07:33 +0000] [1] [INFO] Listening at: http://0.0.0.0:8080 (1)
[2019-11-05 14:07:33 +0000] [1] [INFO] Using worker: sync
[2019-11-05 14:07:33 +0000] [10] [INFO] Booting worker with pid: 10
/usr/local/lib/python3.6/site-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.25.2) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)
[2019-11-05 14:07:35 +0000] [10] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/gunicorn/arbiter.py", line 583, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/base.py", line 129, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/workers/base.py", line 138, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 52, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.6/site-packages/gunicorn/app/wsgiapp.py", line 41, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.6/site-packages/gunicorn/util.py", line 350, in import_app
    __import__(module)
  File "/src/policytool/web/wsgi.py", line 2, in <module>
    from . import api
  File "/src/policytool/web/api.py", line 22, in <module>
    raise Exception("No static directory found. STATIC_ROOT=%r" % STATIC_ROOT)
Exception: No static directory found. STATIC_ROOT='/src/reach/web/build/static'
[2019-11-05 14:07:35 +0000] [10] [INFO] Worker exiting (pid: 10)
[2019-11-05 14:07:35 +0000] [1] [INFO] Shutting down: Master
[2019-11-05 14:07:35 +0000] [1] [INFO] Reason: Worker failed to boot.
```
Spotted on https://github.com/wellcometrust/datalabs-infra/pull/291#issuecomment-549838752

### PR DESCRIPTION
**Pull request by jdu**
# Description

**Note** There are appropriate changes coming to the infra repo in support of these changes as well. As well, as I'm relatively new to the repo, if there's anything I've missed or not considered please let me know and I'll revise.

* Updated file paths to be based in  folder inside containers versus the root of the file system
* Updated gulp watch and gupl build paths to use new paths in
* Updated airflow and other dependents to use  filepaths
* Organized copies into  and  folders under opt for easier debugging inside of containers

This commit mainly consists of changes to various path configurations throughout in order to get a more concrete clean set of directories set up with predictable paths.

The main issue this addresses is that up until this change files, and folders were being mapped into the root of the filesystem which is less than ideal as they are interspersed between system paths/folders making reasonoing about where everything is a bit more difficult than it should be.

A normal standard for how to handle this (one of many) is to use the `/opt` directory on the filesystem to organize applications which are **optional** for the operating system. If we were compiling binaries we might want to move compiled artefacts into `/usr/local/bin` and `/etc/<app_name>` for instance but because we are deploying source code (i.e. python) into the container it makes more sense for it to go into `/opt`.

The other part of this is to make reasoning about paths in the different containers for things further down the line like k8s configuration of ENV variables easier. At first, reading through the Dockerfiles, docker-compose.yml, etc... it was very hard to tell where things were being laid out. Hopefully these changes have made this a little easier and the paths more concrete.

The result of this is that we end up with two main directories under `/opt`

1. `/opt/reach` for application code
2. `/opt/airflow` for airflow config, etc...

The gulp watch runs against `/opt/reach/build/web` 

## Type of change

- [x] :bug: Bug fix

# How Has This Been Tested?

1. Docker tests ran and completed fine.
1. Tested sites working and operational in local dev
1. Tested and ran individual DAG paths to ensure file paths were not effected

# Checklist:

- [x] My code follows the style guidelines of this project (pep8 AND pyflakes)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] If needed, I changed related parts of the documentation
- [ ] I included tests in my PR
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] If my PR aims to fix an issue, I referenced it using `#(issue)`

### CODE REVIEW

**Comment by hblanks** [comment]
@jdu  - I'm off today but can take a look Monday. This should be a good step forward.

Not sure if you while developing this with @SamDepardieu , but review may be easier the next time if you ping one of us on Slack early in the process -- [30% feedback](https://42floors.com/blog/startups/thirty-percent-feedback) can be helpful on these kind of path/name refactors.

**Comment by jdu** [comment]
No problem, I did discuss a bit with @SamDepardieu yesterday just to ask questions around the existing paths (specifically the gulp setup), etc... so he's aware that I was trying to sort it out. Can work on getting feedback in earlier, this mainly came out of looking at addressing #266 and realizing the file paths were a bit all over the place, the changes themselves were only about an hour, but understanding where everything was going in the first place took a bit more time.

This PR is up to get feedback to make sure I haven't missed anything in here, and plan was to do the same on infra to make sure that I'm not missing anything there or kludged up anything.

File: reach/web/gulpfile.js [hunk 1]
```diff
@@ -19,16 +19,16 @@ exports.css = function() {
         require('cssnano')
   ];
 
-  return src('static/**/*.css')
+  return src('src/**/*.css')
```
Comment (SamDepardieu) [inline reach/web/gulpfile.js]: If we change this, we also need to rename `reach/reach/web/static` to `reach/reach/web/src`. Might be good to do that on the same PR, if you don't mind :)

**Comment by hblanks** [comment]
yeah...seems to have just needed a fresh `make web-build-image`...but verifying that gulp is building / watching

**Comment by hblanks** [comment]
OK! Everything verifies fine after rebuilding that target.

**Pull request merged**